plat/mediatek/mt8183/plat_dcm.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <arch.h>
+#include <bakery_lock.h>
+#include <console.h>
+#include <debug.h>
+#include <mmio.h>
+// #include <mt_spm_reg.h>
+#include <plat_dcm.h>
+#include <plat_private.h>
+#include <plat_dcm.h>
+#include <platform.h>
+#include <platform_def.h>
+#include <mtk_plat_common.h>
+
+#ifndef PWR_STATUS /* in mt_spm_reg.h */
+#define PWR_STATUS                     (SPM_BASE + 0x180)
+#endif /* in mt_spm_reg.h */
+
+/* #define DCM_DEBUG */
+
+uint64_t plat_dcm_mcsi_a_addr;
+uint32_t plat_dcm_mcsi_a_val;
+static int plat_dcm_init_type;
+static unsigned int dcm_big_core_cnt;
+int plat_dcm_initiated;
+
+#define PWR_STA_BIG_MP_MASK	(0x1 << 15)
+
+DEFINE_BAKERY_LOCK(dcm_lock);
+
+void dcm_lock_init(void)
+{
+	bakery_lock_init(&dcm_lock);
+}
+
+void dcm_lock_get(void)
+{
+	bakery_lock_get(&dcm_lock);
+}
+
+void dcm_lock_release(void)
+{
+	bakery_lock_release(&dcm_lock);
+}
+
+void plat_dcm_mcsi_a_backup(void)
+{
+}
+
+void plat_dcm_mcsi_a_restore(void)
+{
+}
+
+void plat_dcm_rgu_enable(void)
+{
+}
+
+void plat_dcm_big_core_sync(short on)
+{
+	/* Check if Big cluster power is existed */
+#ifdef DCM_DEBUG
+	console_init(gteearg.atf_log_port, UART_CLOCK, UART_BAUDRATE);
```

Comment:
What are you trying to do here?

If all you want is to output some stuff to the UART at runtime, you can just do

    console_switch_state(CONSOLE_FLAG_BOOT);
    ...print your stuff...
    console_switch_state(CONSOLE_FLAG_RUNTIME);

Since you have this pattern a dozen times, you'll probably want to define a helper function/macro for it (e.g. DCM_DEBUG_PRINT(...) or something).